speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionTupleTests.cpp

```diff
@@ -0,0 +1,99 @@
+#include <gtest/gtest.h>
+#include "TestUtils.h"
+#include "DecisionTuple.h"
+
+#define FLOAT_SET_GET_FUNCTION_TEST(p_setFunction, p_getFunction, p_containsFunction, p_controlValue) \
+    DecisionTuple tuple;                                                                              \
+    tuple.p_setFunction(p_controlValue);                                                              \
+    ASSERT_ALMOST_EQ(tuple.p_getFunction(), p_controlValue, 0.0001f);                                 \
+    ASSERT_TRUE(tuple.p_containsFunction());
+
+#define SET_GET_FUNCTION_TEST(p_setFunction, p_getFunction, p_containsFunction, p_controlValue) \
+    DecisionTuple tuple;                                                                        \
+    tuple.p_setFunction(p_controlValue);                                                        \
+    ASSERT_EQ(tuple.p_getFunction(), p_controlValue);                                           \
+    ASSERT_TRUE(tuple.p_containsFunction());
```

Comment:
change defines to use a third define with the duplicate code

Reply:
using more defines in defines should not be done to not overcomplicate the define